File: src/main/java/no/nav/syfo/kafka/producer/model/KOversikthendelse.java
package no.nav.syfo.kafka.producer.model;

import lombok.Data;
import lombok.experimental.Accessors;

import javax.validation.constraints.*;
import java.time.LocalDateTime;

@Data
@Accessors(fluent = true, chain = true)
public class KOversikthendelse {

    @NotEmpty
    @Pattern(regexp = "^[0-9]{11}$")
    public String fnr;
    @NotEmpty
    public String hendelseId;
    @NotEmpty
    public String enhetId;
    @NotNull
    public LocalDateTime tidspunkt;
}

File: src/main/kotlin/no/nav/syfo/oversikthendelse/KOversikthendelse.kt
package no.nav.syfo.oversikthendelse

import java.io.Serializable
import java.time.LocalDateTime

data class KOversikthendelse(
    val fnr: String,
    val hendelseId: String,
    val enhetId: String,
    val tidspunkt: LocalDateTime
) : Serializable
